client/src/components/Courses.tsx

```diff
@@ -26,23 +34,124 @@ const CourseList = styled.div`
   flex: 3;
 `;
 
-function Courses() {
+interface CoursesProps {
+  coursesAdded?: string[];
+  addCourseToRedux?: any;
+}
+
+function Courses({ coursesAdded, addCourseToRedux }: CoursesProps) {
+  // snackbar:
+  const [open, setOpen] = useState(false);
+  const [message, setMessage] = useState("");
+
+  const [input, setInput] = useState("");
+
+  const handleSnackBarClose = () => {
+    setOpen(false);
+  };
+
+  const handleChange = (e: any) => {
+    setInput(e.target.value);
+  };
+
+  const handleAddBtnClick = async () => {
+    if (input.length === 0) {
+      setMessage("Empty input");
+      setOpen(true);
+      return;
+    }
+    const partition = input.split(" ");
+    if (partition.length !== 2) {
+      setMessage("Invalid format");
+      setOpen(true);
+      return;
+    }
+
+    const department = partition[0].toUpperCase();
+    const courseNumber = partition[1];
+    const response = await axios.get(
+      `http://localhost:5000/api/section/${department}/${courseNumber}`
```

Comment:
Once my PR is merged we can query `http://localhost:5000/api/section/${term}/${department}/${courseNumber}`